force-app/main/default/classes/LeadTriggerHandler.cls

```diff
@@ -102,12 +124,20 @@ public with sharing class LeadTriggerHandler {
 	 * - One of the errors is map related. Make sure you are using the correct contact map key
 	 */
 	public static void handleLeadAutoConvert(List<Lead> leads) {
+
+		if (isRunning) {
+            return;
+        }
+        isRunning = true;
```

Comment:
Are there other ways to check if a lead was converted or not? 
That will stop the recursion from happening.